src/Model/List/ListDecorator.php

```diff
@@ -273,14 +278,14 @@ public function filterByCallback($callback)
     }
 
     /**
-     * @return TList<T>
+     * @return SS_List<T>
      */
-    public function limit(?int $length, int $offset = 0): SS_List&Sortable&Filterable&Limitable
+    public function limit(?int $length, int $offset = 0): SS_List
     {
         return $this->list->limit($length, $offset);
     }
 
-    public function byID($id)
+    public function byID(int|string|null $id): mixed
```

Comment:
```suggestion
    public function byID(mixed $id): mixed
```
Just like with `ArrayList`, we don't know what type of list it is and therefore we don't know what it contains so we don't know what values it might allow for an "ID"

Reply:
We know that it's an SS_List, which is type hinted for `int|string|null $id`?

Reply:
Conversation can be held in https://github.com/silverstripe/silverstripe-framework/pull/11433#discussion_r1815652124 unless the change in https://github.com/silverstripe/silverstripe-framework/pull/11433#discussion_r1815668115 results in reverting it to not be strongly typed.

Reply:
Changed to mixed